File: HR-js/greedy/LuckBalance.java
package greedy;

import java.util.Arrays;
import java.util.Comparator;

public class LuckBalance {

    static void sort2DArray (int[][] contests, int col) {
        Arrays.sort(contests, new Comparator<int[]>() {
            @Override public int compare(int[] e1, int[] e2) {
                if (e1[col] > e2[col])
                    return 1;
                else
                    return -1;
            }
        });
    }

    static int maximizeLuckBalance(int k, int[][] contests) {
        sort2DArray(contests, 0);
        long impContests = Arrays.stream(contests).filter(contest -> contest[1] == 1).count();
        long winContests = impContests - k;
        int lostLuckBal = 0;
        int wonLuckBal = 0;
        int i = 0;
        int winCounter = 0;
        while(i < contests.length) {
            if(contests[i][1] == 1 && winCounter < winContests) {
                lostLuckBal += contests[i][0];
                winCounter++;
            } else {
                wonLuckBal += contests[i][0];
            }
            i++;
        }
        return wonLuckBal - lostLuckBal;
    }

    public static void main(String[] args) {
        int[][] contests = {{5, 1}, {2, 1}, {1, 1}, {8, 1}, {10, 0}, {5, 0}};
        System.out.println(maximizeLuckBalance(3, contests));
    }
}

File: HR-js/greedy/luck-balance.js
function maximize_luck_balance(k, contests) {
    contests.sort((a, b) => a[0] - b[0]);
    const imp_contests = contests.filter((contest) => contest[1] === 1);
    const win_contest = imp_contests.length - k;
    let lost_luck_bal = 0;
    let won_luck_bal = 0;
    let i = 0;
    let win_counter = 0;
    while (i < contests.length) {
        if (contests[i][1] === 1 && win_counter < win_contest) {
            lost_luck_bal += contests[i][0];
            win_counter++;
        } else {
            won_luck_bal += contests[i][0];
        }
        i++;
    }
    return won_luck_bal - lost_luck_bal;
    // console.log(won_luck_bal - lost_luck_bal);
    // console.log(imp_contests.length);
    // console.log(sorted_contests);
}

function main(ip) {
    const ip_arr = ip.split('\n');
    const [n, k] = ip_arr[0].split(' ');
    const contests = []; // luck balance, importance

    for (let i = 1; i <= n; i++) {
        contests.push(ip_arr[i].split(' ').map(Number));
    }

    const res = maximize_luck_balance(k, contests);
    console.log(res);
}

main("6 3\n5 1\n2 1\n1 1\n8 1\n10 0\n5 0");